django/core/urlresolvers.py

```diff
@@ -1,640 +1,10 @@
-                                         key)
-        if ns_pattern:
-            resolver = get_ns_resolver(ns_pattern, resolver)
-
-    return force_text(iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs)))
-
-reverse_lazy = lazy(reverse, six.text_type)
-
-
-def clear_url_caches():
-    get_callable.cache_clear()
-    get_resolver.cache_clear()
-    get_ns_resolver.cache_clear()
-
-
-def set_script_prefix(prefix):
-    """
-    Sets the script prefix for the current thread.
-    """
-    if not prefix.endswith('/'):
-        prefix += '/'
-    _prefixes.value = prefix
-
-
-def get_script_prefix():
-    """
-    Returns the currently active script prefix. Useful for client code that
-    wishes to construct their own URLs manually (although accessing the request
-    instance is normally going to be a lot cleaner).
-    """
-    return getattr(_prefixes, "value", '/')
-
-
-def clear_script_prefix():
-    """
-    Unsets the script prefix for the current thread.
-    """
-    try:
-        del _prefixes.value
-    except AttributeError:
-        pass
-
-
-def set_urlconf(urlconf_name):
-    """
-    Sets the URLconf for the current thread (overriding the default one in
-    settings). Set to None to revert back to the default.
-    """
-    if urlconf_name:
-        _urlconfs.value = urlconf_name
-    else:
-        if hasattr(_urlconfs, "value"):
-            del _urlconfs.value
-
-
-def get_urlconf(default=None):
-    """
-    Returns the root URLconf to use for the current thread if it has been
-    changed from the default one.
-    """
-    return getattr(_urlconfs, "value", default)
-
-
-def is_valid_path(path, urlconf=None):
-    """
-    Returns True if the given path resolves against the default URL resolver,
-    False otherwise.
-
-    This is a convenience method to make working with "is this a match?" cases
-    easier, avoiding unnecessarily indented try...except blocks.
-    """
-    try:
-        resolve(path, urlconf)
-        return True
-    except Resolver404:
-        return False
-
-
-def translate_url(url, lang_code):
-    """
-    Given a URL (absolute or relative), try to get its translated version in
-    the `lang_code` language (either by i18n_patterns or by translated regex).
-    Return the original URL if no translated version is found.
-    """
-    parsed = urlsplit(url)
-    try:
-        match = resolve(parsed.path)
-    except Resolver404:
-        pass
-    else:
-        to_be_reversed = "%s:%s" % (match.namespace, match.url_name) if match.namespace else match.url_name
-        with override(lang_code):
-            try:
-                url = reverse(to_be_reversed, args=match.args, kwargs=match.kwargs)
-            except NoReverseMatch:
-                pass
-            else:
-                url = urlunsplit((parsed.scheme, parsed.netloc, url, parsed.query, parsed.fragment))
-    return url
+warnings.warn(
+    "django.core.urlresolvers is deprecated and will be removed"
```

Comment:
"Importing from django.core.urlresolvers is deprecated in favor of django.urls."
(don't need to mention the version since RemovedInDjango20Warning appears with the message)